addons/sale_product_configurator/static/tests/product_configurator.test.js

```diff
@@ -0,0 +1,264 @@
+                        type: 'many2one',
+                        relation: 'sale_order_line'
+                    }
+                }
+            },
+            sale_product_configurator: {
+                fields: {
+                    product_template_id: {
+                        string: 'product',
+                        type: 'many2one',
+                        relation: 'product_template'
+                    },
+                    product_template_attribute_value_ids: {
+                        type: 'many2many',
+                        relation: 'product_template_attribute_value'
+                    },
+                    product_no_variant_attribute_value_ids: {
+                        type: 'many2many',
+                        relation: 'product_template_attribute_value'
+                    },
+                    product_custom_attribute_value_ids: {
+                        type: 'many2many',
+                        relation: 'product_attribute_custom_value'
+                    }
+                },
+                records: [{
+                    product_template_id: 42
+                }]
+            },
+            product_template_attribute_value: {
+                fields: {
+                    id: {type: 'integer'}
+                }
+            },
+            product_attribute_custom_value: {
+                fields: {
+                    id: {type: 'integer'}
+                }
+            },
+            pricelist: {
+                fields: {
+                    id: {type: 'integer'}
+                }
+            }
+        };
+    }
+}, function (){
+    QUnit.test('Select a non configurable product template and verify that the product_id is correctly set', async function (assert) {
+        assert.expect(2);
+
+        var form = await createView({
+            View: FormView,
+            model: 'sale_order',
+            data: this.data,
+            arch: getArch(),
+            mockRPC: function (route, params) {
+                if (params.method === 'get_single_product_variant') {
+                    assert.ok(true);
+                    return Promise.resolve({product_id: 2});
+                }
+                // FIXME awa: this shouldn't be here since the read is done in 'event_sale'
+                // But at the moment there is no easy way to solve such cross module 'include' issues
+                if (params.method === 'read') {
+                    return Promise.resolve(false);
+                }
+                return this._super.apply(this, arguments);
+            },
+            intercepts: {
+                do_action: function (ev) {
+                    if (ev.data.action === 'sale_product_configurator.sale_product_configurator_action') {
+                        assert.ok(false, "Should not execute the configure action");
+                    }
+                },
+            }
+        });
+
+        await testUtils.dom.click(form.$("a:contains('Add a product')"));
```

Comment:
same as other remarks, there are better helpers for many2one

Reply:
Done.